packages/net-stubbing/lib/server/intercepted-request.ts

```diff
@@ -184,7 +184,31 @@ export class InterceptedRequest {
           }
         }
 
-        if (eventName === 'before:request') {
+        if (
+          eventName === 'before:request' ||
+          // https://github.com/cypress-io/cypress/issues/17139
+          // This `cypressInternalResReplyCall` came
+          // from driver/src/cy/net-stubbing/events/before-requests.ts
+          // to sendStaticResponse at util.ts
+          // to middleware/request.ts
+          // to middleware/response.ts
+          // to here.
+          //
+          // This is a really ugly solution.
+          // But `cypressInternalResReplyCall` is necessary below because of the 3 reasons.
+          // 1. res.reply() call inside responseHandler inside cy.intercept()
+          // doesn't trigger `before:request` event.
+          // 2. We cannot simply add `before:response` below
+          // because it can cause duplicated counting for the same request
+          // (one for `before:request` and one more for `before:response`)
+          // 3. Blocking `before:response` when `before:request` is called doesn't work
+          // because there are multiple `before:response` events for the same request
+          // when you register several intercepts for a single url.
```

Comment:
Can you add test cases for these in addition to/instead of inlining it as a comment? I discovered an issue with this fix while trying to write a test for (3).